controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -88,6 +88,31 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 			response[name] = errors.Wrap(err, "failed to list etcd members using etcd client")
 			continue
 		}
+
+		// Check if any member's node is missing from workload cluster
+		// If any, delete it with best effort
+		for _, m := range members {
+			nodeName := m.Name
+			isFound := false
+			for _, node := range controlPlaneNodes.Items {
+
+				name := node.Name
+				if nodeName == name {
+					isFound = true
+				}
+			}
+
+			if !isFound {
+				if err := w.removeMemberForNode(ctx, nodeName); err != nil {
+					continue
+				}
+
+				if err := w.RemoveNodeFromKubeadmConfigMap(ctx, nodeName); err != nil {
+					continue
+				}
```

Comment:
If the goal is to do a best effort removal, what if we try both and aggregate the errors in a kerrors.NewAggregate error message and log out an info that says we weren't able to delete the node and it could potentially cause issues?